### PR TITLE
Add Emscripten Webassembly build

### PR DESCRIPTION
**Pull request by zekoz**

### CODE REVIEW

File: client/gui-qt/CMakeLists.txt [hunk 1]
```diff
@@ -62,3 +62,9 @@ target_include_directories(gui-qt PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 
 target_link_libraries(gui-qt PRIVATE client)
 target_link_libraries(gui-qt PRIVATE Qt5::Widgets)
+if(APPLE)
+  target_link_libraries(gui-qt PRIVATE "-framework CoreAudio" "-framework ForceFeedback" "-framework AudioToolbox")
+  qt5_import_plugins(gui-qt 
+    INCLUDE Qt5::QCocoaIntegrationPlugin
+  )
+endif()
```
Comment (lmoureaux) [inline client/gui-qt/CMakeLists.txt:70]: Can you move this to #776 ?

File: patches/lua.CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,146 @@
+# Lua can be compiled as either C or C++.
```
Comment (lmoureaux) [inline patches/lua.CMakeLists.txt:1]: What's the license of this file? Any copyright notice we need to keep?

**Review by lmoureaux — CHANGES_REQUESTED** [review]
Code looks reasonable, two simple comments

File: patches/lua.CMakeLists.txt (hunk 2 above)
Comment (zekoz) [reply patches/lua.CMakeLists.txt:1]: MIT license https://github.com/microsoft/vcpkg/blob/master/ports/lua/CMakeLists.txt

Comment (lmoureaux) [reply patches/lua.CMakeLists.txt:1]: Then we need to comply with:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

Comment (zekoz) [reply patches/lua.CMakeLists.txt:27]: Done

Comment (lmoureaux) [reply patches/lua.CMakeLists.txt:27]: Thanks!

**Pull request merged**